packages/api/src/models/Transaction.ts

```diff
@@ -79,6 +81,17 @@ class Transaction extends Base {
   @JoinColumn({ name: 'predicate_id' })
   @ManyToOne(() => Predicate)
   predicate: Predicate;
+
+  static getTypeFromTransactionRequest(transactionRequest: TransactionRequest) {
+    const { type } = transactionRequest;
+    const transactionType = {
+      [FuelTransactionType.Create]: TransactionType.TRANSACTION_CREATE,
+      [FuelTransactionType.Script]: TransactionType.TRANSACTION_SCRIPT,
+      default: TransactionType.TRANSACTION_SCRIPT,
+    };
+
+    return transactionType[type] ?? transactionType.default;
+  }
 }
```

Comment:
This is a utility to obtain the `TransactionType` through a TransactionRequest.

**CreateTransactionRequest**: TransactionType.TRANSACTION_CREATE
**ScriptTransactionRequest**: TransactionType.TRANSACTION_SCRIPT